nestkernel/connection_manager.cpp

```diff
@@ -685,6 +685,10 @@ nest::ConnectionManager::increase_connection_count( const thread tid,
     num_connections_[ tid ].resize( syn_id + 1 );
   }
   ++num_connections_[ tid ][ syn_id ];
+  if ( num_connections_[ tid ][ syn_id ] >= 1 << 27 )
+  {
+    throw KernelException( "Too many connections per thread for synapse." );
```

Comment:
I think this message is not very informative for the end user. How about something like `"Too many connections: at most ??? connections supported per virtual process and synapse model."`? The number can be inserted using `compose()` from `libnestutil`.

Reply:
I have made the error message more informative.